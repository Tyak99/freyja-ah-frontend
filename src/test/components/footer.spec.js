import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Footer from '../../components/footer/footer';

const setUp = (props = {}) => {
  const component = shallow(<Footer {...props} />);
  return component;
};
describe('Footer component', () => {
  let component;
  beforeEach(() => {
    component = setUp();
  });
  it('should render without errors', () => {
    expect(toJson(component)).toMatchSnapshot();
  });
  it('should render children', () => {
    const wrapper = component.find('footer');
    expect(wrapper.find('ul')).toBeTruthy();
  });
  it('should render an unordered list', () => {
    const list = component.find('ul');
    expect(list.length).toBe(1);
    expect(list.find('li').exists()).toBe(true);
  });
  it('should render an ordered list', () => {
    const list = component.find('li');
    expect(list.length).toBe(3);
    expect(list.find('div').exists()).toBe(true);
  });
  it('should render the div', () => {
    const list = component.find('div');
    expect(list.length).toBe(4);
  });
});
