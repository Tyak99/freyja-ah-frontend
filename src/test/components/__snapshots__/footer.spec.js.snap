// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer component should render without errors 1`] = `
<footer
  className="resp"
>
  <ul
    className="footer-nav navvv"
  >
    <li>
      <div>
        How it works     |    
      </div>
    </li>
    <li>
      <div>
        Need help?     |    
      </div>
    </li>
    <li>
      <div>
        Create an account
      </div>
    </li>
  </ul>
  <div
    className="copyright"
  >
    © cslc 2019
  </div>
</footer>
`;
